dhall/src/Dhall/TypeCheck.hs

```diff
@@ -2255,6 +2241,21 @@ prettyTypeMessage (IfBranchMismatch expr0 expr1 expr2 expr3) =
         txt2 = insert expr2
         txt3 = insert expr3
 
+prettyTypeMessage (ListLitInvariant) = ErrorMessages {..}
+  where
+    short = "Invariant: A non-empty list literal must not have a type annotation"
+
+    long =
+        "Explanation: Invariant: A non-empty list literal must not have a type           \n\
```

Comment:
If the user sees this error it's 100% an implementation bug, so the only advice we should give to the user is to report the error and include the offending source span if possible in the error message so that they can copy and paste it